arangod/IResearch/IResearchDataStore.cpp

```diff
@@ -1612,16 +1563,15 @@ bool IResearchDataStore::exists(IResearchDataStore::Snapshot const& snapshot,
       irs::numeric_utils::numeric_traits<LocalDocumentId::BaseType>::raw_ref(
           encoded);
 
-  for (auto const& segment : snapshot.getDirectoryReader()) {
+  for (auto const& segment : reader) {
     auto const* pkField = segment.field(DocumentPrimaryKey::PK());
 
     if (ADB_UNLIKELY(!pkField)) {
       continue;
```

Comment:
Is it actually possible to get here?

Reply:
I don't think so, probably should be assert, but I kept old behavior